examples/daal/cpp_sycl/source/logistic_regression/logistic_regression_online.hpp

```diff
@@ -0,0 +1,219 @@
+/* file: logistic_regression_online.hpp */
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+/*
+!  Content:
+!    C++ example of logistic regression training algorithm implementation for online
+!    computing mode with SGD Minibatch solver and DPC++ interfaces.
+!******************************************************************************/
+
+#include "daal_sycl.h"
+
+namespace daal_dm = daal::data_management;
+namespace daal_al = daal::algorithms;
+namespace daal_solver = daal::algorithms::optimization_solver;
+namespace daal_sgd = daal::algorithms::optimization_solver::sgd;
+
+template <typename Fptype = float>
+class LogisticRegressionOnline
+{
+    using LogLoss = daal_solver::logistic_loss::Batch<Fptype>;
+    using LogLossPtr = daal::services::SharedPtr<LogLoss>;
+
+    using CrossEntropyLoss = daal_solver::cross_entropy_loss::Batch<Fptype>;
+    using CrossEntropyLossPtr = daal::services::SharedPtr<CrossEntropyLoss>;
+
+    using SGDMinibatchSolver = daal_sgd::Batch<Fptype, daal_sgd::miniBatch>;
+    using SGDMinibatchSolverPtr = daal::services::SharedPtr<SGDMinibatchSolver>;
+
+public:
+    LogisticRegressionOnline()
```

Comment:
I would suggest to align with batch

```suggestion
    LogisticRegressionOnline(size_t nClasses)
```

Reply:
Do not see the reason for this. I would like to initialize all the parameters at once instead of separating them.

Reply:
I see you divided on parts: constructors, set_params, set_inputs, compute, finalize by analogy with container of algorithms. So, in this case, it would look consistency. Otherwise, why not pass all these parameters directly to the constructor?

Reply:
It's done with intent to have default ctor of the class - this avoids user the need to create multiple objects to solve the tasks with different parameters, and it's easy to initialize user-owned classes which hold `LogisticRegressionOnline` object - no need to store it by pointer if we have a default ctor.